### PR TITLE
docs(format): add info about filtering and format configuration

### PR DESCRIPTION
**Pull request by dbanksdesign**
*Issue #, if available:* #303 #297

*Description of changes:* Adding documentation in the Formats section around using filters and format configuration. 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: docs/formats.md [hunk 1]
```diff
@@ -31,10 +31,57 @@ You use formats in your config file under platforms > [platform] > files > [file
 
 There is an extensive (but not exhaustive) list of [included formats](#pre-defined-formats) available in Style Dictionary.
 
+### Format configuration
```
Comment (shaunbent) [inline docs/formats.md]: I was just to comment and say this is a great new feature, then looked at my code and fount I am already doing this. 🙈

File: scripts/handlebars/templates/formats.hbs [hunk 2]
```diff
@@ -27,10 +27,57 @@ You use formats in your config file under platforms > [platform] > files > [file
 
 There is an extensive (but not exhaustive) list of [included formats](#pre-defined-formats) available in Style Dictionary.
 
+### Format configuration
+
+Formats are can take configuration to make them more flexible. This allows you to re-use the same format multiple times with different configuration or to allow the format to use data not defined in the tokens themselves. To configure a format add extra attributes on the file object in your configuration like so:
```
Comment (chazzmoney) [inline scripts/handlebars/templates/formats.hbs]: Formats are can take configuration

File: docs/formats.md [hunk 3]
```diff
@@ -31,10 +31,57 @@ You use formats in your config file under platforms > [platform] > files > [file
...
+
+Formats are can take configuration to make them more flexible. This allows you to re-use the same format multiple times with different configuration or to allow the format to use data not defined in the tokens themselves. To configure a format add extra attributes on the file object in your configuration like so:
```
Comment (chazzmoney) [inline docs/formats.md]: Formats are can take configuration

**Comment by chazzmoney** [comment]
:shipit:

**Pull request merged**